### PR TITLE
Don't perform eager loads on aggregate queries

### PR DESCRIPTION
**Pull request by devslashwill**
Fixes an error when using eager loading and aggregate methods on the same query. Previously this would generate a fatalError when attempting to perform the subquery for eager loading.

### CODE REVIEW

File: Sources/FluentKit/Query/QueryBuilder+Paginate.swift [hunk 1]
```diff
@@ -8,8 +8,15 @@ extension QueryBuilder {
     public func paginate(
         _ request: PageRequest
     ) -> EventLoopFuture<Page<Model>> {
-        let count = self.count()
-        let items = self.copy().range(request.start..<request.end).all()
+        // Remove all eager load requests when counting the rows
+        // otherwise we try to read IDs from the count reply when
+        // performing the eager load subqueries.
+        let countQueryBuilder = self.copy()
+        countQueryBuilder.eagerLoads = .init()
```
Comment (tanner0101) [inline Sources/FluentKit/Query/QueryBuilder+Paginate.swift]: Maybe this should happen in `count()` itself? 

For example, if you do: 

```swift
var builder = Planet.query(on: db).with(\.$star)
builder.count()
// later
builder.all()
```

You'd get the same error, right?

Comment (devslashwill) [reply Sources/FluentKit/Query/QueryBuilder+Paginate.swift]: You're correct. I've added it to the aggregate function instead as it's true for all aggregate methods, not just count.

File: Sources/FluentKit/Query/QueryBuilder+Paginate.swift [hunk 2]
```diff
@@ -8,8 +8,8 @@ extension QueryBuilder {
     public func paginate(
         _ request: PageRequest
     ) -> EventLoopFuture<Page<Model>> {
-        let count = self.count()
-        let items = self.copy().range(request.start..<request.end).all()
+        let count = self.copy().count()
+        let items = self.range(request.start..<request.end).all()
```
Comment (tanner0101) [inline Sources/FluentKit/Query/QueryBuilder+Paginate.swift]: This change should be reverted, we need `self.copy().range(...)`, and `count()` already copies internally.

File: Sources/FluentKit/Query/QueryBuilder.swift [hunk 3]
```diff
@@ -564,6 +564,10 @@ public final class QueryBuilder<Model>
         where Result: Codable
     {
         let copy = self.copy()
+        // Remove all eager load requests otherwise we try to
+        // read IDs from the aggreate reply when performing
+        // the eager load subqueries.
+        copy.eagerLoads = .init()
```
Comment (tanner0101) [inline Sources/FluentKit/Query/QueryBuilder.swift]: Perfect 👍

File: Sources/FluentKit/Query/QueryBuilder+Paginate.swift (hunk 2 above)
Comment (tanner0101) [reply Sources/FluentKit/Query/QueryBuilder+Paginate.swift]: Side note, we could write a test for this by doing a paginate followed by an `all()` on the same builder.

**Pull request merged**